blockchain/dkg/dkgtasks/register_task.go

```diff
@@ -71,44 +98,91 @@ func (t *RegisterTask) doTask(ctx context.Context, logger *logrus.Entry, eth int
 		return err
 	}
 
-	if block >= t.State.RegistrationEnd {
-		return errors.Wrapf(objects.ErrCanNotContinue, "At block %v but registration ends at %v", block, t.State.RegistrationEnd)
-	}
+	logger.Info("RegisterTask doTask()")
 
 	// Setup
-	c := eth.Contracts()
-	txnOpts, err := eth.GetTransactionOpts(ctx, t.State.Account)
-	if err != nil {
-		logger.Errorf("getting txn opts failed: %v", err)
-		return err
+	if t.TxOpts == nil {
+		txnOpts, err := eth.GetTransactionOpts(ctx, t.State.Account)
+		if err != nil {
+			logger.Errorf("getting txn opts failed: %v", err)
+			return err
+		}
+
+		nonce, err := eth.GetGethClient().PendingNonceAt(ctx, t.State.Account.Address)
```

Comment:
Dangerous assumption to use pending nonce as truthful state if endpoint is on a loadbalancer and not a single instance.